tools/ci_build/github/azure-pipelines/templates/react-native-ci.yml

```diff
@@ -228,42 +250,32 @@ jobs:
      targetFolder: $(Build.SourcesDirectory)/js/react_native/e2e/android/app/libs
     displayName: Copy Android package to Android e2e test directory
 
-  - template: android-dump-logs-from-steps.yml
-    parameters:
-     steps:
-     - task: Gradle@3
-       inputs:
-         gradleWrapperFile: '$(Build.SourcesDirectory)/js/react_native/e2e/android/gradlew'
-         workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e/android'
-         options: '--stacktrace'
-         tasks: ':app:connectedDebugAndroidTest'
-         publishJUnitResults: true
-         testResultsFiles: '**/TEST-*.xml'
-         testRunTitle: 'React Native Android e2e Test results'
-         javaHomeOption: 'path'
-         jdkDirectory: '$(JAVA_HOME_11_X64)'
-         sonarQubeRunAnalysis: false
-         spotBugsAnalysis: false
-       displayName: Run React Native Android e2e Tests
+  - script: |
+      detox build --configuration android.emu.release
+    workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+    displayName: Build React Native Detox Android e2e Tests
+  
+  - script: |
+      detox test --record-logs all  --configuration android.emu.release
+    workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+    displayName: Run React Native Detox Android e2e Tests
 
   - script: |
-      export FORCE_BUNDLING=1
-      export RCT_NO_LAUNCH_PACKAGER=1
-      export ENTRY_FILE=index.tsx
-      xcrun xcodebuild test -workspace $(Build.SourcesDirectory)/js/react_native/e2e/ios/OnnxruntimeModuleExample.xcworkspace \
-        -scheme OnnxruntimeModuleExample -destination 'platform=iOS Simulator,OS=latest,name=iPhoneRNTest' \
-        -derivedDataPath $(Build.BinariesDirectory)/react_native/ios_e2e_test/derived_data | xcpretty -r junit --no-color \
-        --output $(Build.SourcesDirectory)/js/react_native/e2e/ios/build/reports/test-results.xml
+      detox build --configuration ios.sim.release
     workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
-    displayName: Run React Native iOS e2e tests
+    displayName: Build React Native Detox iOS e2e Tests
+  
+  - script: |
+      detox test --record-logs all --configuration ios.sim.release
+    workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+    displayName: Run React Native Detox iOS e2e Tests
+  
+  - script: |
+      set -e -x
+      ls -R '$(Build.SourcesDirectory)/js/react_native/e2e/artifacts'
+    displayName: Examine detox artifacts/ folder
 
-  - task: PublishTestResults@2
```

Comment:
Not sure how to publish test-results.xml files with `PublishTestResults@2` task here. looks like https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-test-results-v2?view=azure-pipelines&tabs=trx%2Ctrxattachments%2Cyaml#syntax it only supports `JUnit, NUnit, VSTest, XUnit, CTest. ` testRunner styles.

Reply:
perhaps you can get the test results to be output in one of the supported formats. haven't looked into this much, but a quick search turned up https://github.com/jest-community/jest-junit as one possible option.

Reply:
thanks! let me take a look